src/builtins/fish_indent.rs

```diff
@@ -471,6 +481,23 @@
         }
     }
 
+    // Set the formatting mode to on/off; directive_offset is the start index of
+    // the #!fish_indent directive.
+    fn set_formatting(&mut self, on: bool, directive_offset: usize) {
+        match self.fish_indent_off {
+            None if !on => {
+                self.fish_indent_off = Some((directive_offset, self.output.len()));
+            }
+            Some((source_index, output_index)) if on => {
+                // Replace any formatted text we've generated with the original non-formatted text
+                self.output
+                    .replace_range(output_index.., &self.source[source_index..directive_offset]);
```

Comment:
I see, so since we do a single pass through the AST, we don't know
in advance that we don't need to format this?
If it ever becomes a problem in terms of performance,
we can add a cheap first pass to compute the skip ranges.

Reply:
> I see, so since we do a single pass through the AST, we don't know in advance that we don't need to format this? If it ever becomes a problem in terms of performance, we can add a cheap first pass to compute the skip ranges.

I was originally going to wrap all the `self.output.push(...)` bits in an `if self.fish_indent_off.is_none()` or something like that, but I was worried that if someone adds a new AST node in the future, they may forget to wrap the output generating code in an `if`. Whereas this way, they don't need to even know about the `fish_indent: off` feature.